erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -252,17 +255,45 @@ def get_payment_days(self, joining_date, relieving_date):
 
 		return payment_days
 
-	def get_holidays_for_employee(self, start_date, end_date):
+	def get_holidays_for_employee(self, start_date, end_date, get_type=None):
 		holiday_list = get_holiday_list_for_employee(self.employee)
-		holidays = frappe.db.sql_list('''select holiday_date from `tabHoliday`
+		count, d_count = frappe.db.sql('''select count(*),count(distinct holiday_date) from `tabHoliday`
 			where
-				parent=%(holiday_list)s
-				and holiday_date >= %(start_date)s
+				holiday_date >= %(start_date)s
 				and holiday_date <= %(end_date)s''', {
-					"holiday_list": holiday_list,
 					"start_date": start_date,
 					"end_date": end_date
-				})
+				})[0]
+
+		if count == d_count:
```

Comment:
have been thinking about this issue and i think the holiday lists need to be grouped up so you can assign a new holiday list while the old one stays active